src/tracing.ts

```diff
@@ -68,3 +75,59 @@ export function startTracing(opts: Partial<Options> = {}): void {
   // register global provider
   trace.setGlobalTracerProvider(provider);
 }
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function configureHttpInstrumentation(instrumentation: any, options: Options) {
+  if (!options.serverTimingEnabled) {
+    return;
+  }
+
+  if (
+    typeof instrumentation['setConfig'] !== 'function' ||
+    typeof instrumentation['_getConfig'] !== 'function'
+  ) {
+    return;
+  }
+
+  const responseHook: HttpResponseCustomAttributeFunction = (
+    span,
+    response
+  ) => {
+    if (response instanceof ServerResponse) {
+      const { traceId, spanId } = span.context();
+      response.setHeader('Access-Control-Expose-Headers', 'Server-Timing');
+      response.setHeader(
```

Comment:
With the current http instrumentation, the response hook is called before users have access to the response object (https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-plugin-http/src/http.ts#L316)

However if this gets changed in future versions so it's called somewhere later down the line, the header could be overwritten :thinking:

Reply:
Added appending of headers: https://github.com/signalfx/splunk-otel-js/pull/70/commits/ddccfb748d9ea51863dab6e092ef0cee09aedcbb#diff-8e40ce0d3dab91400c9a4d6c3c2577a8222c65dc14f886a2b570b564909bef01R81-R97

Although this will take the first branch 100% of the time, since response is not available for users to modify before the hook is called.